src/Utility/Hash.php

```diff
@@ -370,7 +365,9 @@ protected static function _simpleOp($op, $data, $path, $values = null)
                 }
             } elseif ($op === 'remove') {
                 if ($i === $last) {
-                    unset($_list[$key]);
+                    if (!is_string($_list)) {
```

Comment:
I thought other scalar values can be unset safely and nothing happens, but using `is_array()` would be better, as `_simpleOp('insert')` uses the function [here](https://github.com/chinpei215/cakephp/blob/ea5f05b46136803703b946504bc5d66901f6b36a/src/Utility/Hash.php#L363-L365). And, I could move this check to the first statement of the `elseif ($op === 'remove')` block and return `$data` there. Thanks.